tests/integration/plugins/test_python.py

```diff
@@ -408,3 +411,77 @@ def _get_package_install_commands(self) -> list[str]:
 
     primed_script = lf.project_info.prime_dir / "bin/mytest"
     assert not primed_script.exists()
+
+
+@pytest.mark.slow
+def test_python_plugin_get_files(new_dir, partitions):
+    parts_yaml = textwrap.dedent(
+        """\
+        parts:
+          foo:
+            plugin: python
+            source: .
+            python-packages: [flask==3.1.0]
+        """
+    )
+    parts = yaml.safe_load(parts_yaml)
+
+    lifecycle = LifecycleManager(
+        parts,
+        application_name="test_python",
+        cache_dir=new_dir,
+        partitions=partitions,
+    )
+    actions = lifecycle.plan(Step.BUILD)
+
+    with lifecycle.action_executor() as ctx:
+        ctx.execute(actions)
+
+    part_name = list(parts["parts"].keys())[0]
+    actual_file_list = lifecycle._executor._handler[part_name]._plugin.get_files()
+    part_install_dir = lifecycle._executor._part_list[0].part_install_dir
+
+    # Real quick instantiate another copy of the plugin to ensure statelessness.
+    properties2 = PythonPlugin.properties_class.unmarshal(parts["parts"]["foo"])
+    part_info = PartInfo(
+        project_info=ProjectInfo(
+            application_name="test", cache_dir=new_dir, partitions=partitions
+        ),
+        part=Part("foo", {"source": "."}, partitions=partitions),
+    )
+    plugin2 = PythonPlugin(properties=properties2, part_info=part_info)
+    assert plugin2.get_files() == actual_file_list
+
+    # Make sure all the expected packages were installed.
+    # We can't assert the exact set of keys because the pip version will change
+    # over time.  And we can't assert the number of keys because py3.10 installs
+    # setuptools as a separate package.
+    assert Package(name="Flask", version="3.1.0") in actual_file_list
+
+    # Can't assert specific versions here because flask has >= versions for its
+    # dependencies.
+    seeking_pkgs = {
+        pkgname: False
+        for pkgname in [
+            "Jinja2",
+            "MarkupSafe",
+            "Werkzeug",
+            "blinker",
+            "click",
+            "itsdangerous",
+        ]
+    }
+    for found_pkg in actual_file_list:
+        for sought_pkg in seeking_pkgs:
+            if found_pkg.name == sought_pkg:
+                seeking_pkgs[sought_pkg] = True
+                break
+    sought_collapsed = set(seeking_pkgs.values())
+    success = len(sought_collapsed) == 1 and sought_collapsed.pop()
+    assert success, f"Didn't find one or more expected packages: {seeking_pkgs}"
+
+    # Check a few specifics to make sure we got package contents correctly
+    assert part_install_dir / "bin/flask" in actual_file_list[Package("Flask", "3.1.0")]
+    assert len(actual_file_list[Package("Jinja2", "3.1.4")]) == 57
```

Comment:
This still depends on the specific versions

Reply:
https://github.com/canonical/craft-parts/pull/943/commits/737c0cabd0d368c551ae230c16ffecbd7d46e8de